### PR TITLE
ci: add trivy filesystem scan job

### PR DESCRIPTION
**Pull request by phisco**
Signed-off-by: Philippe Scorsolini <p.scorsolini@gmail.com>

### Description of your changes

Introduce a step running a trivy [filesystem scan](https://aquasecurity.github.io/trivy/v0.19.2/vulnerability/scanning/filesystem/), this will avoid adding dependencies with know fixed vulnerabilities at the time the pipeline is run.

This should be also paired with daily scans against all the supported built images performed either through the registry of choice or a scheduled github action, see [example](https://github.com/goharbor/harbor/blob/main/.github/workflows/nightly-trivy-scan.yml).

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

N.A.

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Review by hasheddan — APPROVED** [review]
Thanks for adding this @phisco!

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -165,6 +165,25 @@ jobs:
...
+          submodules: true
+
+      - name: Run Trivy vulnerability scanner in fs mode
+        uses: aquasecurity/trivy-action@0.8.0
+        with:
+          scan-type: 'fs'
+          ignore-unfixed: true
+          skip-dirs: design
+          scan-ref: '.'
+          severity: 'CRITICAL,HIGH'
```
Comment (hasheddan) [inline .github/workflows/ci.yml:185]: @phisco one question -- do we need deps vendored (similar to what we do for CodeQL) for this to do a full scan? Or will it follow the dependency tree just based on our `go.mod` / `go.sum`?

Comment (phisco) [reply .github/workflows/ci.yml:185]: It scans go.mod/sum, no need to vendor, [here](https://aquasecurity.github.io/trivy/v0.36/docs/vulnerability/languages/golang/) a bit more info!

Comment (hasheddan) [reply .github/workflows/ci.yml:185]: Thanks!

**Pull request merged**